### PR TITLE
Add support for building with libuv rather than the native event loop

### PR DESCRIPTION
**Pull request by tgoyne**
Mostly just for the sake of being able to actually test it.

### CODE REVIEW

File: src/util/node/event_loop_signal.hpp [hunk 1]
```diff
@@ -30,8 +29,6 @@ class EventLoopSignal {
 
         // This assumes that only one thread matters: the main thread (default loop).
         uv_async_init(uv_default_loop(), m_handle, [](uv_async_t* handle) {
-            // The v8::Local handles need a "scope" to be present or will crash.
-            Nan::HandleScope scope;
```
Comment (tgoyne) [inline src/util/node/event_loop_signal.hpp]: The rational for this removal is that it's v8-specific while the rest of this file works with libuv on any platform, so it should be done specifically in the JS binding.

Comment (bdash) [inline src/util/node/event_loop_signal.hpp]: Should the directory this file is in be renamed from `node` to `uv` to better reflect that?

**Comment by bdash** [comment]
👍

**Comment by bmunkholm** [comment]
@tgoyne Who else needs to review this if any? Please ping them or merge.
/cc @jpsim

**Pull request merged**